core/state_processor.go

```diff
@@ -122,5 +124,11 @@ func ApplyTransaction(config *params.ChainConfig, bc *BlockChain, author *common
 	receipt.Logs = statedb.GetLogs(tx.Hash())
 	receipt.Bloom = types.CreateBloom(types.Receipts{receipt})
 
+	retData := types.ReturnData{TxHash: receipt.TxHash, Data: data, Removed: false}
+	txPostEvent := TransactionEvent{TxHash: receipt.TxHash, RetData: retData}
+	if bc != nil { // For some tests unrelated to tx events, this ptr can be nil; should always be non-nil in production
+		go bc.txPostFeed.Send(&txPostEvent)
+	}
```

Comment:
I don't know whether the slowdown would be significant, but as it turns out I have to change this anyway.  In order to handle chain reorgs, I've realized these events need to all get posted after the entire block is done, not individually after each one.  I'm rewriting that already, so that this function returns the data and then it gets posted by the caller at the end of the block.